mllib/src/test/scala/org/apache/spark/ml/feature/QuantileDiscretizerSuite.scala

```diff
@@ -91,6 +44,46 @@ class QuantileDiscretizerSuite
       "Observed number of buckets does not equal expected number of buckets.")
   }
 
+  test("Test transform method on unseen data") {
+    val sqlCtx = SQLContext.getOrCreate(sc)
+    import sqlCtx.implicits._
+
+    val trainDF = sc.parallelize(1.0 to 100.0 by 1.0).map(Tuple1.apply).toDF("input")
+    val testDF = sc.parallelize(-10.0 to 110.0 by 1.0).map(Tuple1.apply).toDF("input")
+    val discretizer = new QuantileDiscretizer()
+      .setInputCol("input")
+      .setOutputCol("result")
+      .setNumBuckets(5)
+
+    val result = discretizer.fit(trainDF).transform(testDF)
+    val firstBucketSize = result.filter(result("result") === 0.0).count
+    val lastBucketSize = result.filter(result("result") === 4.0).count
+
+    assert(firstBucketSize === 30L,
+      "Size of first bucket did not equal expected value.")
+    assert(lastBucketSize === 31L,
+      "Size of last bucket did not equal expected value.")
+  }
+
+  test("Test default relativeError and setRelativeError") {
+    val discretizer = new QuantileDiscretizer()
+    val testValues = Array(0.1, 0.2, 0.3, 0.4)
```

Comment:
minor, but having these values is not really testing much. Perhaps `Array(0.0, 0.005, 0.1)` will be better, to test that we can set an explicit value below, between and above the default range.